x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/connector_types/microsoft_defender_endpoint.ts

```diff
@@ -0,0 +1,260 @@
+        .on('error', logErrorDetails)
+        .send({
+          name: 'My sub connector',
+          connector_type_id: MICROSOFT_DEFENDER_ENDPOINT_CONNECTOR_ID,
+          config: {
+            clientId: 'client-abc',
+            tenantId: 'tenant-123',
+            oAuthServerUrl: 'https://some.non.existent.com',
+            oAuthScope: 'scope-a',
+            apiUrl: 'https://some.non.existent.com',
+          },
+          secrets: { clientSecret: 'abc-123' },
+        })
+        .expect(200);
+
+      connectorId = response.body.id;
+    });
+
+    after(async () => {
+      if (connectorId) {
+        await supertest
+          .delete(`${getUrlPrefix('default')}/api/actions/connector/${connectorId}`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(({ ok, status }) => {
+            // Should cover all success codes (ex. 204 (no content), 200, etc...)
+            if (!ok) {
+              throw new Error(
+                `Expected delete to return a status code in the 200, but got ${status}`
+              );
+            }
+          });
+
+        connectorId = '';
+      }
+    });
+
+    describe('Sub-action authz', () => {
+      describe('and user has NO privileges', () => {
+        let user: CreatedUser;
+
+        before(async () => {
+          user = await createUser({
+            username: 'read_access_user',
+            kibanaFeatures: { actions: ['read'] },
+          });
+        });
+
+        after(async () => {
+          if (user) {
+            await user.deleteUser();
+          }
+        });
+
+        for (const subActionValue of subActions) {
+          it(`should deny execute of ${subActionValue}`, async () => {
+            const execRes = await executeSubAction({
+              supertest: supertestWithoutAuth,
+              subAction: subActionValue,
+              subActionParams: {},
+              username: user.username,
+              password: user.password,
+              expectedHttpCode: 403,
+              errorLogger: logErrorDetails.ignoreCodes([403]),
+            });
+
+            expect(execRes.body).to.eql({
+              statusCode: 403,
+              error: 'Forbidden',
+              message: `Unauthorized to execute a "${MICROSOFT_DEFENDER_ENDPOINT_CONNECTOR_ID}" action`,
+            });
+          });
+        }
+      });
+
+      describe('and user has proper privileges', () => {
+        let user: CreatedUser;
+
+        before(async () => {
+          user = await createUser({
+            username: 'all_access_user',
+          });
+        });
+
+        after(async () => {
+          if (user) {
+            await user.deleteUser();
+            // @ts-expect-error
+            user = undefined;
+          }
+        });
+
+        for (const subActionValue of subActions) {
+          const isAllowedSubAction =
+            subActionValue === MICROSOFT_DEFENDER_ENDPOINT_SUB_ACTION.TEST_CONNECTOR;
+
+          it(`should ${
```

Comment:
FYI: these tests are already setup to work on top of https://github.com/elastic/kibana/pull/203503, so they will fail for now